telemetry/src/types.rs

```diff
@@ -0,0 +1,41 @@
+/// Encapsulates the retry period that is calculated as `min_period + 2 ^ min(exponent, max_exponent)`
```

Comment:
Let's rename file to `retry_period.rs`. This would be more close to domain driven design. I would suggest to not group modules by `types`, `const` or other cases, instead we should focus on their function and place in our design.